Maestro/src/Microsoft.DotNet.Maestro.WebApi/Services/VerifyPayloadSignatureAttribute.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http.Controllers;
+using System.Web.Http.Filters;
+
+namespace Microsoft.DotNet.Maestro.WebApi.Services
+{
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class, Inherited = true)]
+    public class VerifyPayloadSignatureAttribute : FilterAttribute, IAuthorizationFilter
+    {
+        private const string SignatureHeaderName = "X-Hub-Signature";
+
+        public async Task<HttpResponseMessage> ExecuteAuthorizationFilterAsync(
+            HttpActionContext actionContext,
+            CancellationToken cancellationToken,
+            Func<Task<HttpResponseMessage>> continuation)
+        {
+            using (Task<byte[]> getRequestContentTask = actionContext.Request.Content.ReadAsByteArrayAsync())
+            {
+                try
+                {
+                    string payloadSignature = GetPayloadSignature(actionContext.Request);
+                    string computedHash = ComputeHashHexDigest(await getRequestContentTask);
+
+                    if (!CryptographicEquals(payloadSignature, computedHash))
+                    {
+                        Trace.TraceWarning($"Payload signature: {payloadSignature}");
+                        Trace.TraceWarning($"Computed hash: {computedHash}");
+
+                        throw new Exception($"The payload signature specified by the {SignatureHeaderName} " +
```

Comment:
Why not just TraceError here, and then return the Unauthorized response?  Instead of throwing an exception and then catching it, just to return the response.